### PR TITLE
Create homework4.py

### PR DESCRIPTION
**Pull request by eugenesamozdran**

### CODE REVIEW

File: homework/jenya_s/homework4.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+def find_most_frequent(text):
+
+    # making a new string in lowercase
+    # and replacing punctuation marks in it with spaces
+    sub_str = text.lower()
+    translation_table = dict.fromkeys(map(ord, ",.:;!?-"), " ")
+    sub_str2 = sub_str.translate(translation_table)
+
+    # creating a list of separate words and getting rid of spaces
+    lst = [x for x in sub_str2.split(" ") if x!=""]
```
Comment (aodarc) [inline homework/jenya_s/homework4.py]: this can be simple `[x for x in sub_str2.split(" ") if x]`

File: homework/jenya_s/homework4.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    
+    # forming a list of most frequent words
+    final_lst = [i for i in lst if lst.count(i) >= (max(lst.count(i) for i in lst))]
```
Comment (aodarc) [inline homework/jenya_s/homework4.py]: move max value calculating into separate variable, it decreases complexity of expression

Comment (aodarc) [inline homework/jenya_s/homework4.py]: use `set comprehension`

File: homework/jenya_s/homework4.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    # removing duplicates
+    final_lst = list(set(final_lst))
```
Comment (aodarc) [inline homework/jenya_s/homework4.py]: use `sorted` function

File: homework/jenya_s/homework4.py (hunk 1 above)
Comment (eugenesamozdran) [reply homework/jenya_s/homework4.py]: if x!="" part was added specifically for cases, when several punctuation marks are added in a row(just like in the example "TOm?!"). The matter is that in such cases some additional spaces are present in a list of separate words, you may test it from your side if you want.

**Comment by eugenesamozdran** [comment]
Please check the updated file

File: homework/jenya_s/homework4.py (hunk 1 above)
Comment (aodarc) [reply homework/jenya_s/homework4.py]: you ca remove `!=""` part and this will be the same

**Pull request merged**